### ISSUE TITLE
Support for slick 3.0

### ISSUE THREAD

**Issue by mgosk** [VirtusLab/unicorn#45]
Please add support for slick 3.0 
http://slick.typesafe.com/news/2014/12/19/slick-3.0.0-M1-released.html

**Comment by Kwestor** [comment]
Needs to be updated to RC2: http://slick.typesafe.com/news/2015/03/27/slick-3.0.0-RC2-released.html

**Comment by bambuchaAdm** [comment]
Ok. After slick is went on `Future` path we need to have `Database` instance on repository. Now all method are warped with `Future`. Work related to #61 is go on branch `feature/slick-3.0`. Tests need to be upgraded too, but `unicorn-core` is pass compilation phase and could be used for testing. Happy hacking.

**Comment by jvican** [comment]
Thanks for your work @bambuchaAdm.
Could you please upload the branch feature/slick-3.0 in a public repository? I would like to access it via sbt. I tried to isolate the unicorn-core project but I got weird errors of dependency with play-slick even when build.sbt and Dependencies.scala have been modified.
I would love to have access via maven or any help.

**Comment by bambuchaAdm** [comment]
Branch [feature/slick-3.0](https://github.com/VirtusLab/unicorn/tree/feature/slick-3.0) is on this repo. Just clone it and checkout on branch. I have backported one of compilation fixes form main branch to slick-3.0. Thanks for interesting with unicorn .

**Comment by jvican** [comment]
The final version of slick-3.0 is out. Can you upgrade and upload unicorn to mvn central?

**Comment by Kwestor** [comment]
Yes, but tests were not migrated, I have to fix them first.

**Comment by javierfvargas** [comment]
Do you have an estimate date of release for slick 3.0 support?

**Comment by jirihelmich** [comment]
Is there some document describing what exactly needs to be done? I would like to help finishing this, since the branch https://github.com/VirtusLab/unicorn/tree/v0.7.x-slick-3.0.x seems to be quite dead... But I have no idea what needs to be done.

I would love to migrate my project to Play 2.4 + Slick 3.0, but using Unicorn prevents me from doing so...

Kind regards,
Jirka

**Comment by Kwestor** [comment]
We have to change return types from Future to Actions (see #64). Then, all tests have to be updated to work with async.

**Comment by mgilbertnz** [comment]
Hi guys,

I really like this library but I wanted to use with Play 2.4 so I've changed the return types from Futures to Actions.  I've also modified the tests so they now all pass.

I would like to create a pull request for someone to review and make sure I'm on the right track and then once the code looks good then you guys are welcome to merge it if you like.  At the risk of asking a stupid question.  How do I create a pull request when I don't own the repository?

Cheers,
Matt
